crates/ruff/src/checkers/ast/analyze/expression.rs

```diff
@@ -441,7 +441,11 @@ pub(crate) fn expression(expr: &Expr, checker: &mut Checker) {
                 pyupgrade::rules::redundant_open_modes(checker, call);
             }
             if checker.enabled(Rule::NativeLiterals) {
-                pyupgrade::rules::native_literals(checker, expr, func, args, keywords);
+                pyupgrade::rules::native_literals(
+                    checker,
+                    call,
+                    checker.semantic().current_expression_parent(),
```

Comment:
Since we already pass in `checker`, might more straightforward to do this call in the rule itself?

Reply:
Well, I wasn't sure if the rule should assume the expression it's been given is the current one and it seemed safer to push that assumption up to the visitor since it _knows_ that's the case. I'm happy to move it if we tend to make that assumption though!